assets/js/blocks/product-gallery/frontend.tsx

```diff
@@ -41,10 +47,57 @@ interface Store {
 	ref?: HTMLElement;
 }
 
+interface Event {
+	keyCode: number;
+}
+
 type SelectorsStore = Pick< Store, 'context' | 'selectors' | 'ref' >;
 
 interactivityApiStore( {
 	state: {},
+	effects: {
+		woocommerce: {
+			keyboardAccess: ( store: Store ) => {
+				const { context, actions } = store;
+				let allowNavigation = true;
+
+				const handleKeyEvents = ( event: Event ) => {
+					if (
+						! allowNavigation ||
+						! context.woocommerce.isDialogOpen
+					) {
+						return;
+					}
+
+					// Disable navigation for a brief period to prevent spamming
+					allowNavigation = false;
+
+					setTimeout( () => {
+						allowNavigation = true;
+					}, 300 );
```

Comment:
Do you think we should use `requestAnimationFrame` here instead of setTimeout? This would ensure that the navigation is re-enabled as soon as possible without causing any unnecessary delays. Please correct me if my approach isn't right. 

```suggestion
					requestAnimationFrame( () => {
						allowNavigation = true;
					} );
```

Reply:
Thanks for the suggestion. I never knew that method existed. TIL.